uvicorn/config.py

```diff
@@ -446,6 +448,9 @@ def load(self) -> None:
                 ca_certs=self.ssl_ca_certs,
                 ciphers=self.ssl_ciphers,
             )
+
+        elif self.ssl_context_factory:
+            self.ssl = self.ssl_context_factory()
```

Comment:
Hmm... Maybe we shouldn't ignore the previous parameters passed? :thinking:

What I propose was for the factory function to receive a `SSLContext`, which would be the one that `uvicorn` creates on the lines above.